src/analysis/ASTnodefunctions.ts

```diff
@@ -29,120 +27,218 @@ import { FieldWeight, TypeWeightObject, Variables } from '../@types/buildTypeWei
+            }
+        }
+        return complexity + maxFragmentComplexity;
     }
-    return complexity;
-}
 
-export function definitionNode(
-    node: DefinitionNode,
-    typeWeights: TypeWeightObject,
-    variables: Variables
-): number {
-    let complexity = 0;
-    // check the kind property against the set of definiton nodes that are possible
-    if (node.kind === Kind.OPERATION_DEFINITION) {
-        // check if the operation is in the type weights object.
-        if (node.operation.toLocaleLowerCase() in typeWeights) {
-            // if it is, it is an object type, add it's type weight to the total
-            complexity += typeWeights[node.operation].weight;
-            // console.log(`the weight of ${node.operation} is ${complexity}`);
-            // call the function to handle selection set node with selectionSet property if it is not undefined
-            if (node.selectionSet) {
-                complexity += selectionSetNode(
-                    node.selectionSet,
-                    typeWeights,
-                    variables,
-                    node.operation
-                );
+    definitionNode(node: DefinitionNode): number {
+        let complexity = 0;
+        // check the kind property against the set of definiton nodes that are possible
+        if (node.kind === Kind.OPERATION_DEFINITION) {
+            // check if the operation is in the type weights object.
+            if (node.operation.toLocaleLowerCase() in this.typeWeights) {
+                // if it is, it is an object type, add it's type weight to the total
+                complexity += this.typeWeights[node.operation].weight;
+                // console.log(`the weight of ${node.operation} is ${complexity}`);
+                // call the function to handle selection set node with selectionSet property if it is not undefined
+                if (node.selectionSet) {
+                    complexity += this.selectionSetNode(node.selectionSet, node.operation);
+                }
             }
+        } else if (node.kind === Kind.FRAGMENT_DEFINITION) {
+            // Fragments can only be defined on the root type.
+            // Parse the complexity of this fragment once and store it for use when analyzing other
+            // nodes. The complexity of a fragment can be added to the selection cost for the query.
+            const namedType = node.typeCondition.name.value;
+            // Duplicate fragment names are not allowed by the GraphQL spec and an error is thrown if used.
+            const fragmentName = node.name.value;
+
+            if (this.fragmentCache[fragmentName]) return this.fragmentCache[fragmentName];
+
+            const fragmentComplexity = this.selectionSetNode(
+                node.selectionSet,
+                namedType.toLowerCase()
+            );
+
+            this.fragmentCache[fragmentName] = fragmentComplexity;
+            return complexity; // Don't count complexity here. Only when fragment is used.
```

Comment:
This return statement isn't necessary. The function will return on line 225 regardless. Functions with multiple return statements in different places can get hard to follow.